### DISCUSSION TITLE
getListing details result in 400 with lang: 'en'

### DISCUSSION THREAD

**Discussion by creharmony** [etsy/open-api#915] [Q&A] [answered]
hi,

I would like to know if I'm alone with this usecase. 

since this afternoon, on [getListing](https://developers.etsy.com/documentation/reference/#operation/getListing) details with `lang=en` I got 400 as result

🟢Example OK 200

https://openapi.etsy.com/v3/application/listings/887146454?includes=Images%2CTranslations%2CInventory&language=fr

🔴 Example 400 

https://openapi.etsy.com/v3/application/listings/887146454?includes=Images%2CTranslations%2CInventory&language=en
Result:
```
{
  message: 'Request failed with status code 400',
  code: 'ERR_BAD_REQUEST'
}
```

best regards

**Comment by nateswart** [comment]
Earlier today we [released a fix](https://github.com/etsy/open-api/releases/tag/3.0.0-general-release-2023-03-16) for a series of translation issues. All endpoints that return a Listing (not just getListing) will now default the language to the shop’s default language.

Also, specifically on getListing you can override this behavior by passing in a language parameter. If the Listing does not support the language that was requested we return a 400 error with a message indicating the Listing translation does not exist.

To make sure this operates properly, we recommend using the full language code; in this case `en-US` should work for you.

**Comment by creharmony** [comment]
thanks for your reply, I didn't see relase.

I tried the following query params 
````
no language set // OK (response is french)
language: 'fr' // OK
language: 'en'  // 400
language: 'en-US'  // 400
language: 'en-GB'  // 400 // pick from portal source code
language: 'en-UK'  // 400
language: 'de'  // 400
language: 'oo-UU' // 400 // just curious :)
````

so strange that you mention this listing didn't have english translation, 
because when switching in english language via etsy website, I got it with english description. 
I never had translation issue before. And we've got 'en' 400 response for each listing.

in my example, etsy listing is : https://www.etsy.com/fr/listing/887146454/bougie-detente-et-apaisement-bougie


When editing the listing under etsy portal, there is a section dedicated to the translation to replace the default etsy translation for English (title, description, tag). But in the past we left it blank (we always left it blank for all listings) and it worked for English but doesn't seem to work anymore after the release.

**Reply by kniels21** [reply]
Hi,
  A fix has been deployed to revert back to the previous `language` param functionality. You should no longer receive a `400` even if the language translation is not supported by your Listing. You will instead fallback to `en` if an english translation is defined or the shop default language. This fix also corrects the introduced issue where machine language translations were not able to be used via the `language` override. Thanks for reporting this issue.

**Comment by creharmony** [comment, marked as answer]
I added to my previous test (before revert) a column with "post revert" result `aka // rever:`
 ✅ it's OK for `en` for me 👍

 Thanks for this action, I mark this as answer 🙏 

````
       // https://developers.etsy.com/documentation/reference/#operation/getListing
       const includes = "Images,Translations,Inventory";
       var query;
       query =  { includes };                     // OK in french // revert: OK in french
       query =  { includes, language: 'fr' };     // OK in french // revert: OK in french
       query =  { includes, language: 'en' };     // 400          // revert: OK in english
       query =  { includes, language: 'en-US' };  // 400          // revert: OK in english
       query =  { includes, language: 'en-GB' };  // 400          // revert: OK in english
       query =  { includes, language: 'en-UK' };  // 400          // revert: 400
       query =  { includes, language: 'oo-UU' };  // 400          // revert: 400
       query =  { includes, language: 'de' };     // 400          // revert: OK in german

````
